Intersections_3/include/CGAL/Intersections_3/internal/intersection_3_1_impl.h

```diff
@@ -1651,29 +1679,130 @@ inline bool
 do_intersect(const Plane_3<R> &plane1, const Plane_3<R> &plane2,
              const Plane_3<R> &plane3, const R& r)
 {
-  return bool(intersection(plane1, plane2, plane3, r));
+  typedef typename R::RT RT;
+
+  if(! is_zero(determinant(plane1.a(), plane1.b(), plane1.c(),
+                           plane2.a(), plane2.b(), plane2.c(),
+                           plane3.a(), plane3.b(), plane3.c()))){
+    return true;
+  }
+
+  int pcount = 0;
+  bool b12, b13,b23;
+  if(b12 = parallel(plane1,plane2)) pcount++;
```

Comment:
warning: using the result of an assignment as a condition without parentheses [-Wparentheses]